### DISCUSSION TITLE
Using specific local action

### DISCUSSION THREAD

**Discussion by dev0110** [community/community#45351] [Actions] [answered]
### Select Topic Area

Question

### Body

Hi, I'm trying to set up a Workflow that uses a specific action in my local repository. I was following this tutorial: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-an-action-in-the-same-repository-as-the-workflow

My current setup for the directories is the .github folder, followed by two subfolders (actions and workflows). Inside of actions, I have an action.yml and in the workflows I have a main.yml. In my main.yml, if I do 

- name: Checkout
uses: actions/checkout@v3
- name: Custom action
uses: ./.github/actions

then the workflow runs without issues, but if I do  

- name: Checkout
uses: actions/checkout@v3
- name: Custom action
uses: ./.github/actions/action

then I get: Can't find 'action.yml', 'action.yaml', or 'Dockerfile' under '/home/runner/work/repo-name/-repo-name/.github/actions/action'.

Am I misunderstanding the tutorial? It seems like the last part of the path shouldn't be a single file but rather another folder (actions/action/action.yml - for example) - but I'm not sure if that's 100% correct. Thanks for any clarification!

**Comment by dev0110** [comment]
I was taking a look at another post which seems to confirm my last part of the post, but I just wasn't sure if I could specifically point the action I want to use instead of nesting it in another subfolder

**Comment by airtower-luna** [comment, marked as answer]
Each action needs its own folder, you can't have two actions in one folder. Inside the action folder must be an `action.yml` or `action.yaml` file with the metadata for that action. The path set for `uses` must be the one to the folder.

So if you want to have two actions under `.github/actions/`, create a subfolder for each.

**Reply by dev0110** [reply]
Thank you! I was trying to see if I could just have several actions under 1 folder, but that makes more sense.
